### PR TITLE
Added check for symmetry of input adjacency matrix

### LINKED ISSUE #4493: Check for symmetry needed in hidden_linear_function.py for input matrix A
### What is the expected enhancement?
The input adjacency matrix A for `HiddenLinearFunction` should be required to be a symmetric matrix. A check for the same seems to be missing in `qiskit/circuit/library/hidden_linear_function.py`

Please let me know if I'm missing something here.

### PR DESCRIPTION
**Pull request by MadhavJivrajani**
### Summary
Fixes #4493 
Added a check for the adjacency matrix being symmetric in nature to `hidden_linear_function.py` in `qiskit/circuit/library`

### Details and comments
Imported CircuitError to check for symmetry of the adjacency matrix.

Deleted tests added for hidden_linear_function.py in a previous commit, will be adding more comprehensive tests soon.

### CODE REVIEW

File: qiskit/circuit/library/hidden_linear_function.py [hunk 1]
```diff
@@ -80,6 +81,10 @@ def __init__(self,
             adjacency_matrix: a symmetric n-by-n list of 0-1 lists.
                 n will be the number of qubits.
         """
+        adjacency_matrix = np.array(adjacency_matrix)
+        if not np.allclose(adjacency_matrix, adjacency_matrix.transpose()):
+            raise CircuitError("The adjacency matrix must be symmetric in nature.")
```
Comment (Cryoris) [inline qiskit/circuit/library/hidden_linear_function.py:86]:
```suggestion
            raise CircuitError("The adjacency matrix must be symmetric.")
```

File: qiskit/circuit/library/hidden_linear_function.py [hunk 2]
```diff
@@ -80,6 +81,10 @@ def __init__(self,
             adjacency_matrix: a symmetric n-by-n list of 0-1 lists.
                 n will be the number of qubits.
         """
+        adjacency_matrix = np.array(adjacency_matrix)
```
Comment (Cryoris) [inline qiskit/circuit/library/hidden_linear_function.py:84]: Using `asarray` avoids copying the matrix.
```suggestion
        adjacency_matrix = np.asarray(adjacency_matrix)
```

**Comment by MadhavJivrajani** [comment]
I have committed the suggested changes. 

Style lint check fails saying:
```
qiskit/circuit/library/hidden_linear_function.py:76:4: W9006: "CircuitError" not documented as being raised (missing-raises-doc)
```

Any pointers on how I could go about fixing it?

**Comment by Cryoris** [comment]
Add
```
Raises:
    CircuitError: If A is not symmetric.
```
to the class docstring.

**Comment by MadhavJivrajani** [comment]
Thanks!

**Comment by Cryoris** [comment]
Could you add a test for this as well? Something like
```python
with self.assertRaises(CircuitError):
   HLF(non-symmetric-matrix)
```

**Comment by MadhavJivrajani** [comment]
Should I send it in with the HLF unitary test?

**Comment by Cryoris** [comment]
As you prefer. If you'll send it with #4492, then we can only merge it once this is one merged, but that's fine, too.

**Comment by MadhavJivrajani** [comment]
Alright great, I'll send it in with #4492, thanks for all the help!

**Pull request merged**